packages/client/src/schema/query.ts

```diff
@@ -364,18 +365,18 @@ export class Query<Record extends XataRecord, Result extends XataRecord = Record
       console.trace('Calling getMany does not return all results. Paginate to get all results or call getAll.');
     }
 
-    const array = new RecordArray(page, results.slice(0, size));
+    const array = new PageRecordArray(page, results.slice(0, size));
 
     // Method overloading does not provide type inference for the return type.
-    return array as unknown as RecordArray<Result>;
+    return array as unknown as PageRecordArray<Result>;
   }
 
   /**
    * Performs the query in the database and returns all the results.
    * Warning: If there are a large number of results, this method can have performance implications.
    * @returns An array of records from the database.
    */
-  getAll(): Promise<Result[]>;
+  getAll(): Promise<RecordArray<Result>>;
```

Comment:
This is the main change that needed to happen right? What was the motivation for creating `PageRecordArray`?